### ISSUE TITLE
Incompatable scala version

### ISSUE THREAD

**Issue by zhichao-li** [intel-analytics/BigDL#341]
Currently the scala version in bigdl is fixed to be  ``` <scala.version>2.10.5</scala.version> ``` which might cause issue while running under Spark2.0 which use scala 2.11 by default.
```
Exception in thread "main" java.lang.NoSuchMethodError: scala.reflect.api.JavaUniverse.runtimeMirror(Ljava/lang/ClassLoader;)Lscala/reflect/api/JavaMirrors$JavaMirror;
        at com.intel.analytics.bigdl.example.imageclassification.ImagePredictor$$anonfun$main$1.apply(ImagePredictor.scala:76)
        at com.intel.analytics.bigdl.example.imageclassification.ImagePredictor$$anonfun$main$1.apply(ImagePredictor.scala:41)
        at scala.Option.map(Option.scala:146)
        at com.intel.analytics.bigdl.example.imageclassification.ImagePredictor$.main(ImagePredictor.scala:41)
        at com.intel.analytics.bigdl.example.imageclassification.ImagePredictor.main(ImagePredictor.scala)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.apache.spark.deploy.SparkSubmit$.org$apache$spark$deploy$SparkSubmit$$runMain(SparkSubmit.scala:729)
        at org.apache.spark.deploy.SparkSubmit$.doRunMain$1(SparkSubmit.scala:185)
        at org.apache.spark.deploy.SparkSubmit$.submit(SparkSubmit.scala:210)
        at org.apache.spark.deploy.SparkSubmit$.main(SparkSubmit.scala:124)

**Comment by zhichao-li** [comment]
@yiheng @yansh0625

**Comment by yiheng** [comment]
spark 2.0 use scala 2.11.8 and spark 1.6 use scala 2.10.5. I think we can make this line changable: https://github.com/intel-analytics/BigDL/blob/master/pom.xml#L113 
in our 2.0 profile

**Comment by jason-dai** [comment]
closed via #351
